File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <string>
#include <deque>

using namespace std;

int main()
{
	int n, val;
	cin >> n;

	deque<int>D;
	string cmd;
	while (n--)
	{
		ios::sync_with_stdio(0);
		cin.tie(0);

		cin >> cmd;

		if (cmd == "push_front") {
			cin >> val;
			D.push_front(val);
		}
		else if (cmd == "push_back") {
			cin >> val;
			D.push_back(val);
		}
		else if (cmd == "pop_front") {
			if (D.empty()) cout << -1 << '\n';
			else {
				cout << D.front() << '\n';
				D.pop_front();
			}
		}
		else if (cmd == "pop_back") {
			if (D.empty()) cout << -1 << '\n';
			else {
				cout << D.back() << '\n';
				D.pop_back();
			}
		}
		else if (cmd == "size") {
			cout << D.size() << '\n';
		}
		else if (cmd == "empty") {
			cout << D.empty() << '\n';
		}
		else if (cmd == "front") {
			if (D.empty()) cout << -1 << '\n';
			else cout << D.front() << '\n';
		}
		else if (cmd == "back") {
			if (D.empty()) cout << -1 << '\n';
			else cout << D.back() << '\n';
		}
	}

}

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		Deque<String> deque = new ArrayDeque<>();
		
		while(n-- > 0){
			StringTokenizer st = new StringTokenizer(br.readLine());
			String cmd = st.nextToken();
			if(cmd.equals("push_front")){
				deque.offerFirst(st.nextToken());
			}else if(cmd.equals("push_back")){
				deque.offerLast(st.nextToken());
			}else if(cmd.equals("pop_front")){
				if(deque.isEmpty()) System.out.println(-1);
				else System.out.println(deque.removeFirst());
			}else if(cmd.equals("pop_back")){
				if(deque.isEmpty()) System.out.println(-1);
				else System.out.println(deque.removeLast());
			}else if(cmd.equals("size")){
				System.out.println(deque.size());
			}else if(cmd.equals("front")){
				if(deque.isEmpty()) System.out.println(-1);
				else System.out.println(deque.peek());
			}else if(cmd.equals("back")){
				if(deque.isEmpty()) System.out.println(-1);
				else System.out.println(deque.peekLast());
			}else if(cmd.equals("empty")){
				if(deque.isEmpty()) System.out.println(1);
				else System.out.println(0);
			}
		}
		
	}
}